sdk/eventhub/Azure.Messaging.EventHubs/MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,393 @@
+ - [Migrating code from `PartitionReceiver` to `EventHubConsumerClient` for reading events in batches](#migrating-code-from-partitionreceiver-to-eventhubconsumerclient-for-reading-events-in-batches)
+ - [Migrating code from `EventProcessorHost` to `EventProcessorClient` for receiving events](#migrating-code-from-eventprocessorhost-to-eventprocessorclient-for-reading-events)
+
+### Migrating code from `PartitionSender` to `EventHubProducerClient` for publishing events to a partition
+
+In v4, events could be published to a single partition using `PartitionSender`. You could also send a single event, a set of events, or an explicit batch. We are only supporting batch currently to ensure that there are no unexpected exceptions generated during send; you can't put it in a batch if it was too large to send.
+
+In v5, this has been consolidated into a more efficient SendAsync(EventDataBatch) method. Batching merges information from multiple events into a single publish message, reducing the amount of network communication needed vs publishing events one at a time. Events are published to a specific partition when partition id is set in [`CreateBatchOptions`](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.createbatchoptions?view=azure-dotnet-preview) before calling [`CreateBatchAsync(createBatchOptions)`](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventhubproducerclient.createbatchasync?view=azure-dotnet-preview#Azure_Messaging_EventHubs_EventHubProducerClient_CreateBatchAsync_Azure_Messaging_EventHubs_CreateBatchOptions_System_Threading_CancellationToken_).
+
+The code below assumes all events fit into a single batch. For more complete example, see sample: [Publish events
+to specific partition](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample06_PublishAnEventBatchToASpecificPartition.cs).
+
+In v4:
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var connectionStringBuilder = new EventHubsConnectionStringBuilder(connectionString){ EntityPath = eventHubName }; 
+var eventHubClient = EventHubClient.CreateFromConnectionString(connectionStringBuilder.ToString());
+PartitionSender partitionSender  =  eventHubClient.CreatePartitionSender("my-partition-id");
+try
+{
+    EventDataBatch eventBatch = partitionSender.CreateBatch();
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("First")));
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Second")));
+
+    await partitionSender.SendAsync(eventBatch);
+}
+catch()
+{
+    // Handle the exception here
+}
+finally
+{
+    await eventHubClient.CloseAsync();
+    await partitionSender.CloseAsync();
```

Comment:
By convention, we should.   In reality, the client will take responsibility and close the receiver, since it spawned it.   Probably a good change to make, though.

Reply:
As Jesse said, client closes all the senders and receivers that are bound to that client but I accidentally closed the client first I'll update the code.